CMakeLists.txt

```diff
@@ -0,0 +1,163 @@
+# uriparser - RFC 3986 URI parsing library
+#
+# Copyright (C) 2018, Sebastian Pipping <sebastian@pipping.org>
+# All rights reserved.
+#
+# Redistribution  and use in source and binary forms, with or without
+# modification,  are permitted provided that the following conditions
+# are met:
+#
+#     * Redistributions   of  source  code  must  retain  the   above
+#       copyright  notice, this list of conditions and the  following
+#       disclaimer.
+#
+#     * Redistributions  in  binary  form must  reproduce  the  above
+#       copyright  notice, this list of conditions and the  following
+#       disclaimer   in  the  documentation  and/or  other  materials
+#       provided with the distribution.
+#
+#     * Neither  the name of the <ORGANIZATION> nor the names of  its
+#       contributors  may  be  used to endorse  or  promote  products
+#       derived  from  this software without specific  prior  written
+#       permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT  NOT
+# LIMITED  TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND  FITNESS
+# FOR  A  PARTICULAR  PURPOSE ARE DISCLAIMED. IN NO EVENT  SHALL  THE
+# COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL,    SPECIAL,   EXEMPLARY,   OR   CONSEQUENTIAL   DAMAGES
+# (INCLUDING,  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES;  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+# HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT  LIABILITY,  OR  TORT (INCLUDING  NEGLIGENCE  OR  OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+# OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+cmake_minimum_required(VERSION 3.0)
+
+project(uriparser
+    VERSION
+        0.9.1
+)
+
+include(GNUInstallDirs)
+
+option(BUILD_SHARED_LIBS "build shared libraries (rather than static ones)" true)
+
+#
+# config.h (only needed by testrunner)
+#
+set(PACKAGE_VERSION ${PROJECT_VERSION})  # for configure_file
```

Comment:
Since you've declared a project with `project(uriparser VERSION)` you already have a PROJECT_VERSION defined and also a uriparser_VERSION. Could be probably better to expose those variables in config.h?

Reply:
Simplified now, thanks!